### DISCUSSION TITLE
How to specify the font encoding correctly

### DISCUSSION THREAD

**Discussion by yang-521** [pymupdf/PyMuPDF#3506] [Looking for help]
I try to specify the encoding of the font, but the output files are actually Identity-H

    font = fitz.Font(fontname="宋体",fontfile="images/simsun.ttc")
    for page in doc:
        page.insert_font(fontname="宋体", fontbuffer=font.buffer, encoding='utf-8')  
        page.insert_text(
            position,
            text,
            fontname="宋体",
            fontsize=font_size,
            encoding='utf-8' 
        )

**Comment by JorjMcKie** [comment]
If you only want to use `insert_text` or `insert_textbox` all you need is `page.insert_font(fontname="your-choice", fontfile=file_path)`.
Then use `page.insert_text(point, text, fontname="your-choice",...)`.
You cannot influence the encoding here.

**Reply by yang-521** [reply]
How can I add a custom encoded font? This is the result I added with Acrobat
![image](https://github.com/pymupdf/PyMuPDF/assets/122206917/e987f559-c6f6-4b4f-adb4-c57fa0cb8603)

**Comment by yang-521** [comment]
Can I change the built-in font

**Reply by JorjMcKie** [reply]
You cannot.

**Comment by JorjMcKie** [comment]
But you can use a program like FontForge https://fontforge.org/en-US/ and change some font with it.

**Comment by yang-521** [comment]
> You cannot.

Can non-embedded fonts be added

**Reply by JorjMcKie** [reply]
No, not supported.
